### PR TITLE
feat: add stencil sink

### PR DESCRIPTION
**Pull request by scortier**
Currently supporting schema format :

- [x] JSON
- [x] AVRO

### CODE REVIEW

File: plugins/sinks/stencil/sink.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+		Type:       JsonType(resource.GetType()),
+		Properties: properties,
+	}
+
+	return record, nil
+}
+
+func (s *Sink) buildJsonProperties(table *assetsv1beta1.Table) (properties []map[string]Property) {
+	columns := table.GetSchema().GetColumns()
+	if columns == nil {
```
Comment (StewartJingga) [inline plugins/sinks/stencil/sink.go:160]: let's do `len(columns) > 0` instead. Nil slice will return 0 for length.

File: plugins/sinks/stencil/sink.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			dataType = JsonTypeArray
+		case "object":
+			dataType = JsonTypeObject
+		case "bool", "boolean":
+			dataType = JsonTypeBoolean
+		default:
+			dataType = JsonTypeString
+		}
+
+		columnRecord[column.Profile.String()] = Property{
```
Comment (StewartJingga) [inline plugins/sinks/stencil/sink.go:183]: we will only need `columnRecord[column.Name]`, we don't need the others.

File: plugins/sinks/stencil/sink.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+		Schema:     "https://json-schema.org/draft/2020-12/schema",
+		Title:      resource.GetName(),
+		Type:       JsonType(resource.GetType()),
+		Properties: properties,
+	}
+
+	return record, nil
+}
+
+func (s *Sink) buildJsonProperties(table *assetsv1beta1.Table) (properties []map[string]Property) {
```
Comment (StewartJingga) [inline plugins/sinks/stencil/sink.go:158]: it should only return `map[string]Property`.

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: plugins/sinks/stencil/sink_test.go [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+						Description: "It is the user ID",
+					},
+					"description": {
+						Type:        []stencil.JsonType{stencil.JsonTypeString, stencil.JsonTypeNull},
+						Description: "It is the description",
+					},
+				},
+			},
+		},
+		//{
```
Comment (ravisuhag) [inline plugins/sinks/stencil/sink_test.go:169]: @scortier If this piece of code is not needed, let's remove this instead of commenting.

Comment (scortier) [reply plugins/sinks/stencil/sink_test.go:169]: Sure, will add more test and refactor remaining part, working on it

**Pull request merged**